prometrics.go

```diff
@@ -0,0 +1,75 @@
+package prometrics
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type Registry struct {
+	reg prometheus.Registerer
+	gat prometheus.Gatherer
+}
+
+func NewRegistry() *Registry {
+	r := prometheus.NewRegistry()
+	r.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+	r.MustRegister(prometheus.NewGoCollector())
+	return &Registry{
+		reg: r,
+		gat: r,
+	}
+}
+
+func (r *Registry) ReportDuration() func(*gin.Context) {
+	duration := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "request_duration_seconds",
+			Help:    "Display duration by code, method, and route",
+			Buckets: prometheus.DefBuckets,
+		},
+		[]string{"code", "method", "route"},
+	)
+	r.reg.MustRegister(duration)
+
+	return func(c *gin.Context) {
+		start := time.Now()
+		c.Next()
+		delta := time.Now().Sub(start)
+
+		// create labels
+		labels := prometheus.Labels{}
+		labels["code"] = strconv.Itoa(c.Writer.Status())
+		labels["method"] = c.Request.Method
+		labels["route"] = c.FullPath()
+		duration.With(labels).Observe(delta.Seconds())
+	}
+}
+
+func (r *Registry) ReportConcurrentReq() func(*gin.Context) {
```

Comment:
It could be useful to watch concurrent requests by some label, e.g. by method and route. This could be expanded to accommodate this, or to handle arbitrary labels, or to push that into a exported function altogether.